### ISSUE TITLE
Counter64 OID Values Are Returned as nil

### ISSUE THREAD

**Issue by ichekaldin** [swisscom/ruby-netsnmp#24]
This is a follow up to #23. Apologies for not being clear previously.

For reference, here is my setup.

Ruby:
```
$ ruby -v
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-linux]
```

Ubuntu:
```
$ cat /etc/issue
Ubuntu 16.04.4 LTS \n \l
```

ruby-netsnmp Gem:
```
$ gem query netsnmp

*** LOCAL GEMS ***

netsnmp (0.1.5)
```

The values of Counter64 OIDs are reported as nil. Compare the following.

Here is an example of `ifHCInOctets` table returned by snmpwalk:
```
$ snmpwalk -v3 -l authNoPriv -u myuser -a sha -A "mypassword" myhost.example.com ifHCInOctets
IF-MIB::ifHCInOctets.1 = Counter64: 92491218
IF-MIB::ifHCInOctets.2 = Counter64: 16155773819
IF-MIB::ifHCInOctets.3 = Counter64: 0
IF-MIB::ifHCInOctets.15 = Counter64: 14546980809
IF-MIB::ifHCInOctets.17 = Counter64: 15662965
IF-MIB::ifHCInOctets.18 = Counter64: 11646536
IF-MIB::ifHCInOctets.19 = Counter64: 12299679
IF-MIB::ifHCInOctets.20 = Counter64: 122748726
IF-MIB::ifHCInOctets.21 = Counter64: 12830296
```

And here is the same table values returned by Sensu plugin using ruby-netsnmp gem:
```
$ ruby metrics-snmp-if.rb -h myhost.example.com -u myuser -p mypassword -l auth_no_priv | grep ifHCInOctets
snmp.interfaces.lo.ifHCInOctets
snmp.interfaces.eth0.ifHCInOctets
snmp.interfaces.vti90.ifHCInOctets
snmp.interfaces.vti91.ifHCInOctets
snmp.interfaces.vti0.ifHCInOctets
snmp.interfaces.vti1.ifHCInOctets
snmp.interfaces.vti2.ifHCInOctets
snmp.interfaces.vti3.ifHCInOctets
```

I believe the simple fix is to modify `convert_application_asn` method in lib/netsnmp/varbind.rb from:
```
    def convert_application_asn(asn)
      case asn.tag
      when 0 # IP Address
        IPAddr.new_ntoh(asn.value)
      when 1, # ASN counter 32
           2 # gauge
        val = asn.value
        val.prepend("\x00") while val.bytesize < 4
        asn.value.unpack("N*")[0] || 0
```
to:
```
    def convert_application_asn(asn)
      case asn.tag
      when 0 # IP Address
        IPAddr.new_ntoh(asn.value)
      when 1, # ASN counter 32
           2, # gauge
           6  # ASN Counter 64
        val = asn.value
        val.prepend("\x00") while val.bytesize < 4
        asn.value.unpack("N*")[0] || 0
```

Example of my output after the fix is applied:
```
$ ruby metrics-snmp-if.rb -h myhost.example.com -u myuser -p mypassword -l auth_no_priv | grep ifHCInOctets
snmp.interfaces.lo.ifHCInOctets 92491218 1530802973
snmp.interfaces.eth0.ifHCInOctets 63110204 1530802973
snmp.interfaces.vti90.ifHCInOctets 56824152 1530802973
snmp.interfaces.vti91.ifHCInOctets 15665023 1530802973
snmp.interfaces.vti0.ifHCInOctets 11648594 1530802973
snmp.interfaces.vti1.ifHCInOctets 12301529 1530802973
snmp.interfaces.vti2.ifHCInOctets 122750576 1530802973
snmp.interfaces.vti3.ifHCInOctets 12832458 1530802973
```

**Comment by HoneyryderChuck** [comment]
Nice, so we have a fix! Can I ask you to provide a test and (maybe) add some test for the Counter32 and Counter64 case?

**Comment by ichekaldin** [comment]
You may need to help me with the syntax as this is a bit beyond my Ruby skills.

I think the test could look something like this:

```
        it "decrypts counter32" do
          counter = "\x00\xC4\x1Eg"
          varbind = described_class.new(".1.3.6.1.2.1.31.1.1.1.6.21", type: :counter32, value: counter)
          expect(varbind.convert_application_asn).to eq(12852839)
        end
        
        it "decrypts counter64" do
          counter = "\x00\xC4\x1Eg"
          varbind = described_class.new(".1.3.6.1.2.1.31.1.1.1.6.21", type: :counter64, value: counter)
          expect(varbind.convert_application_asn).to eq(12852839)
        end
```

In other words: value of `"\x00\xC4\x1Eg"` should be converted to `12852839` for both Counter32 and Counter64.

**Comment by HoneyryderChuck** [comment]
sure, you can count on my help!

You can add conversion specs in [varbind_spec.rb](https://github.com/swisscom/ruby-netsnmp/blob/master/spec/varbind_spec.rb#L19-L28), make also sure that conversions correctly encode.

**Comment by ichekaldin** [comment]
Just submitted #25. I believe that does the trick.

**Comment by HoneyryderChuck** [comment]
Just released `0.1.6`.

**Comment by ichekaldin** [comment]
Great, thank you!
